src/CFamily.UnitTests/CMake/CompilationDatabaseLocatorTests.cs

```diff
@@ -155,28 +160,47 @@ public void Locate_HasCMakeSettingsFile_ReturnsConfiguredPathIfItExists(bool fil
         }
 
         [TestMethod]
-        [DataRow("c:\\absolute\\", "c:\\absolute\\")]
-        [DataRow("relative", "relative")]
-        [DataRow("c:\\absolute\\${name}\\${projectDir}", "c:\\absolute\\my-config\\dummy root")]
-        [DataRow("c:\\${projectDir}\\somefolder\\${name}\\sub", "c:\\dummy root\\somefolder\\my-config\\sub")]
-        [DataRow("${projectDir}\\somefolder\\${name}\\sub", "dummy root\\somefolder\\my-config\\sub")]
-        public void Locate_HasCMakeSettingsFile_ConfiguredPathHasParameters_ParametersReplaced(string configuredPath, string expectedPath)
+        public void Locate_MacroEvaluatorIsCalled_EvaluatedPropertyIsNull_Null()
         {
-            var configProvider = CreateConfigProvider("my-config");
-            var cmakeSettings = CreateCMakeSettings("my-config", configuredPath);
-            var cmakeSettingsProvider = CreateCmakeSettingsProvider(
-                new CMakeSettingsSearchResult(cmakeSettings, "", ""));
+            var context = new MacroEvalContext(null, unevaluatedBuildRoot: "property that can't be evaluated");
 
-            var compilationDatabaseFullLocation = Path.GetFullPath(Path.Combine(expectedPath, CompilationDatabaseLocator.CompilationDatabaseFileName));
-            
-            var fileSystem = new Mock<IFileSystem>();
-            fileSystem.SetFileExists(compilationDatabaseFullLocation);
+            var result = context.TestSubject.Locate();
 
-            var testSubject = CreateTestSubject(RootDirectory, configProvider, cmakeSettingsProvider.Object, fileSystem.Object);
+            result.Should().BeNull();
+            context.MacroEvalService.Invocations.Count.Should().Be(1);
+            context.Logger.AssertOutputStringExists(string.Format(Resources.UnableToEvaluateBuildRootProperty, "property that can't be evaluated"));
+        }
 
-            var result = testSubject.Locate();
+        [TestMethod]
+        [DataRow(null, null)]
+        [DataRow("c:\\absolute", "c:\\absolute\\compile_commands.json")]
+        [DataRow("c:\\absolute\\", "c:\\absolute\\compile_commands.json")]
+        [DataRow("c:\\path", "c:\\path\\compile_commands.json")]
+        [DataRow("c:\\path\\", "c:\\path\\compile_commands.json")]
+        [DataRow("c:\\aaa\\..\\bbb", "c:\\bbb\\compile_commands.json")]  // non-canonical path
+        public void Locate_MacroEvaluatorIsCalled_RootedPath_ExpectedValueIsReturn(string macroEvalReturnValue, string expectedResult)
```

Comment:
This test and the next one are actually testing that the macro evaluator is called and that the returned value is processed correctly i.e. combine with file name and put in a canonical form. I didn't think it was worth splitting the tests further.